subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -541,6 +541,11 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 		idx++;
 	}
 
+	if ((idx == 1) && (iface_mode == WIFI_MODE_AP)) {
```

Comment:
shouldn't the check be `argc == 1`?  If a user gives `wifi ap enable <testing> <psk>` then it will fails as `idx == 1` will be hit, no?

Reply:
Channel is mandatory for AP mode, so if user forget channel it should fail. psk is will optional as there maybe a open security. I think idx == 1 should be right check for AP mode.